plugin/plaintextmodel.cpp

```diff
@@ -264,6 +273,26 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detect(const char *encodedCharacters, int length)
+{
+    QTextCodec *codec = QTextCodec::codecForLocale();
+
+    uchardet_t cd = uchardet_new();
+    int res = uchardet_handle_data(cd, encodedCharacters, length);
+    if(!res)
+    {
+        uchardet_data_end(cd);
+
+        const char * name = uchardet_get_charset(cd);
+        codec = QTextCodec::codecForName(name);
```

Comment:
Need to check, but hope the name strings are compatible.

Reply:
Well, I tried ICU at first, but it detects non-latin text with errors. Btw there is the same case in jolla-notes. Thats why I had to find another charset detector. I

> This isn't really just russian charset, but any non-utf8. Furthermore russian per se should already work, provided it's utf-8, right?

Yes. Russian text with UTF-8 encoding works, but if we have file with russian text with Windows-1251 encoding, it opend with non-human-readable text. The same situation is also with arabic text.